# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../webmin/nodefuser.patch
# Copyright (C) 2004 - 2009 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur webmin-1.490-orig/setup.sh webmin-1.490/setup.sh
--- webmin-1.490-orig/setup.sh	2009-12-01 20:21:51.000000000 +0000
+++ webmin-1.490/setup.sh	2009-12-01 20:27:38.000000000 +0000
@@ -452,7 +452,6 @@
 			echo "Webmin does not support being started at boot time on your system."
 		fi
 	fi
-	makeboot=$atboot
 
 	# Copy files to target directory
 	echo "***********************************************************************"
@@ -469,8 +468,8 @@
 	echo "Creating web server config files.."
 	cfile=$config_dir/miniserv.conf
 	echo "port=$port" >> $cfile
-	echo "root=$wadir" >> $cfile
-	echo "mimetypes=$wadir/mime.types" >> $cfile
+	echo "root=/$prefix" >> $cfile
+	echo "mimetypes=/$prefix/mime.types" >> $cfile
 	echo "addtype_cgi=internal/cgi" >> $cfile
 	echo "realm=Webmin Server" >> $cfile
 	echo "logfile=$var_dir/miniserv.log" >> $cfile
@@ -514,15 +513,7 @@
 	md5pass=`$perl -e 'print crypt("test", "\\$1\\$A9wB3O18\\$zaZgqrEmb9VNltWTL454R/") eq "\\$1\\$A9wB3O18\\$zaZgqrEmb9VNltWTL454R/" ? "1\n" : "0\n"'`
 
 	ufile=$config_dir/miniserv.users
-	if [ "$crypt" != "" ]; then
-		echo "$login:$crypt:0" > $ufile
-	else
-		if [ "$md5pass" = "1" ]; then
-			$perl -e 'print "$ARGV[0]:",crypt($ARGV[1], "\$1\$XXXXXXXX"),":0\n"' "$login" "$password" > $ufile
-		else
-			$perl -e 'print "$ARGV[0]:",crypt($ARGV[1], "XX"),":0\n"' "$login" "$password" > $ufile
-		fi
-	fi
+	touch $ufile
 	chmod 600 $ufile
 	echo "userfile=$ufile" >> $cfile
 
@@ -559,11 +550,7 @@
 	echo "Creating access control file.."
 	afile=$config_dir/webmin.acl
 	rm -f $afile
-	if [ "$defaultmods" = "" ]; then
-		echo "$login: $allmods" >> $afile
-	else
-		echo "$login: $defaultmods" >> $afile
-	fi
+	touch $afile
 	chmod 600 $afile
 	echo "..done"
 	echo ""
@@ -584,7 +571,7 @@
 echo "Creating start and stop scripts.."
 rm -f $config_dir/stop $config_dir/start $config_dir/restart $config_dir/reload
 echo "#!/bin/sh" >>$config_dir/start
-echo "echo Starting Webmin server in $wadir" >>$config_dir/start
+echo "echo Starting Webmin server in $prefix" >>$config_dir/start
 echo "trap '' 1" >>$config_dir/start
 echo "LANG=" >>$config_dir/start
 echo "export LANG" >>$config_dir/start
@@ -595,13 +582,13 @@
 echo "export PERLLIB" >>$config_dir/start
 uname -a | grep -i 'HP/*UX' >/dev/null
 if [ $? = "0" ]; then
-	echo "exec '$wadir/miniserv.pl' $config_dir/miniserv.conf &" >>$config_dir/start
+	echo "exec '/opt/webmin/miniserv.pl' $config_dir/miniserv.conf &" >>$config_dir/start
 else
-	echo "exec '$wadir/miniserv.pl' $config_dir/miniserv.conf" >>$config_dir/start
+	echo "exec '/opt/webmin/miniserv.pl' $config_dir/miniserv.conf" >>$config_dir/start
 fi
 
 echo "#!/bin/sh" >>$config_dir/stop
-echo "echo Stopping Webmin server in $wadir" >>$config_dir/stop
+echo "echo Stopping Webmin server in $prefix" >>$config_dir/stop
 echo "pidfile=\`grep \"^pidfile=\" $config_dir/miniserv.conf | sed -e 's/pidfile=//g'\`" >>$config_dir/stop
 echo "kill \`cat \$pidfile\`" >>$config_dir/stop
 
